.github/workflows/pull-request.yml

```diff
@@ -2,9 +2,9 @@ name: PR Testing
 
 on:
   pull_request:
-  push:
-    branches:
-      - master
+    types:
+      - opened
+      - synchronize
```

Comment:
It is default behaviour for pull_request.
I think, we should keep tests on push to master, since if you merge with merge commit, check won't be run for master HEAD commit (just created merge commit)